qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -48,13 +48,26 @@ class SparseArray:  # type: ignore
 
 
 class SamplerQNN(NeuralNetwork):
-    """A Neural Network implementation based on the Sampler primitive."""
+    """A Neural Network implementation based on the Sampler primitive.
+
+    The ``Sampler QNN`` is a neural network that takes in a parametrized quantum circuit
+    with the combined network's feature map (input parameters) and ansatz (weight parameters)
+    and outputs its measurements for the forward and backward passes.
+    The output can be set up in different formats, and an optional post-processing step
+    can be used to interpret the sampler's output in a particular context (e.g. mapping the
+    resulting bitstring to match the number of classes).
+
+    Attributes:
+
+        sampler (BaseSampler): The sampler primitive used to compute the neural network's results.
+        gradient (BaseSamplerGradient): An optional sampler gradient to be used for the backward pass.
+    """
 
     def __init__(
         self,
         *,
-        sampler: BaseSampler,
-        circuit: QuantumCircuit,
+        sampler: BaseSampler | None = None,
+        circuit: QuantumCircuit | None = None,
```

Comment:
Docstrings ought to state  what happens with None ie. like for sampler a reference Sampler() instance is used.

From what I can tell having None for a circuit though will fail below so to me None is invalid.